runtime/src/set_auth_manifest.rs

```diff
@@ -401,6 +415,34 @@ impl SetAuthManifestCmd {
             soc_ifc,
         )?;
 
+        // Sort the image metadata list by firmware ID in place. Also check for duplicate firmware IDs.
+        let slice =
+            &mut metadata_mailbox.image_metadata_list[..metadata_mailbox.entry_count as usize];
+        for i in 0..slice.len() {
+            for j in 0..(slice.len() - 1 - i) {
+                // This check is needed to avoid out of bounds access panic.
+                if (j + 1) >= slice.len() {
+                    break;
+                }
+
+                match slice[j].fw_id.cmp(&slice[j + 1].fw_id) {
+                    core::cmp::Ordering::Greater => {
+                        slice.swap(j, j + 1);
```

Comment:
If you use insertion sort instead (just changing the indices a little), then this runs something like 4x faster.

Reply:
Updated. Also added unit tests for this.